src/Http/Http.Extensions/src/HttpJsonOptionsSetup.cs

```diff
@@ -0,0 +1,24 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization.Metadata;
+using Microsoft.AspNetCore.Http.Json;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.Http;
+
+public sealed class HttpJsonOptionsSetup : IConfigureOptions<JsonOptions>
+{
+    [UnconditionalSuppressMessage("Trimmer", "IL2026", Justification = "Calls System.Text.Json.Serialization.Metadata.DefaultJsonTypeInfoResolver.DefaultJsonTypeInfoResolver().")]
+    public void Configure(JsonOptions options)
+    {
+        if (options.SerializerOptions.TypeInfoResolver is null)
+        {
+            options.SerializerOptions.TypeInfoResolver = new DefaultJsonTypeInfoResolver();
+
+            // TODO: Should we make it readonly????
+            options.SerializerOptions.MakeReadOnly();
```

Comment:
If we wanted to do this, it would need to be "last"

Reply:
If we not do this we will never use typeinfo cache until a call to the Serializer.(De)Serialize. 

https://github.com/dotnet/runtime/blob/91af060b086ab83ba3200fbc7049854900b08523/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs#L56